metaseq/checkpoint_utils.py

```diff
@@ -47,11 +47,7 @@ def save_checkpoint(
         best_function = max if cfg.maximize_best_checkpoint_metric else min
         save_checkpoint.best = best_function(val_loss, prev_best)
 
-    if cfg.no_save:
```

Comment:
--no-save is still useful for benchmarking. To this end, the warning is still helpful.

Reply:
So, should I add `--no-save` back in?

Reply:
Can we gate this to depend on the `--benchmark` flag then?

Reply:
I changed benchmark in opt_baseline now to set `--save-interval` and `--save-interval-updates` to 0, which I felt is cleaner than adding an additional flag just for this use case. If you feel different, I can add a new flag back as well.